Gobabygo/Gobabygo.ino

```diff
@@ -2,58 +2,84 @@ const int joyStickXPin (0);
 const int joyStickYPin (1);
 const int rightMotorPin (6);
 const int leftMotorPin (5);
-const float deadzone = 0.1;
 
 unsigned long currentTime = 0;
 unsigned long previousTime = 0;
 
+float joyXVal = 0;
+float previousJoyXVal = 0;
+float joyYVal = 0;
+float previousJoyYVal = 0;
+
+
 // This function will scale a value between -1 and 1 according to a deadzone:
-void correctForDeadzone(float axis, float deadzone) {
+float correctForDeadzone(float axis, float deadzone) {
   if (abs(axis) <= deadzone){
-    axis = 0;
+    return 0;
+  }
+  else if (axis > deadzone){
+    return (axis - 1)/(1 - deadzone) + 1; // modified point-slope form of a line
   }
   else {
-    if (axis > deadzone) {
-      axis = (axis - 1)/(1 - deadzone) + 1; // modified point-slope form of a line
-    }
-    else {
-      axis = (axis + 1)/(1 - deadzone) - 1;
-    }
+    return (axis + 1)/(1 - deadzone) - 1;
   }
 }
 
+
+// This funtion gradually accelerates a value toward a target
+float accelerateToValue(float currentVal, float targetVal, int timeElapsed, float acceleration) {
+  return currentVal + (targetVal - currentVal) * timeElapsed * acceleration;
+}
+
+
 void setup() {
   // put your setup code here, to run once:
   Serial.begin(9600);
   pinMode(rightMotorPin, OUTPUT);
   pinMode(leftMotorPin, OUTPUT);
 }
 
+
 void loop() {
   // put your main code here, to run repeatedly:
-  float joyXVal;
-  float scaledJoyXVal;
-  int joyYVal;
-  float scaledJoyYVal;
-  
   float rightMotorVal;
   float leftMotorVal;
   float scaler;
+
+  int timeSincePrint;
+  int timeSinceAcceleration;
```

Comment:
It’s best to initialize these values to 0 for clarity. Arduino may have default initialization but not all programming languages do so it’s a good habit to define an initial value